### PR TITLE
Metadata validation: Batch Call Graph API to validate Microsoft alias

### PR DESCRIPTION
**Pull request by LowEntropyBody**
Related issue: #4605

The previous one by one logic made it very slow to verify Microsoft alias. Stress testing has been passed, such changes will not cause exceptions.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4865)

### CODE REVIEW

**Comment by LowEntropyBody** [comment]
Work item: https://dev.azure.com/ceapex/Engineering/_workitems/edit/100605

File: src/docfx/lib/msgraph/MicrosoftGraphCache.cs [hunk 1]
```diff
@@ -43,57 +41,52 @@ public MicrosoftGraphCache(Config config)
...
+        public async Task<(Error error, MicrosoftAlias msAlias)> GetMicrosoftAliasAsync(string alias)
         {
             if (string.IsNullOrEmpty(alias))
             {
-                return new Task<(Error error, MicrosoftAlias msAlias)>(null, null);
+                return default;
             }
 
-            return Synchronized(GetAsyncCore);
+            if (_aliases.TryGetValue(alias, out var msAlias))
```
Comment (OsmondJiang) [inline src/docfx/lib/msgraph/MicrosoftGraphCache.cs]: concurrentdictionary.GetOrAdd()

File: src/docfx/lib/log/TelemetryName.cs [hunk 2]
```diff
@@ -11,5 +11,6 @@ internal enum TelemetryName
         LoadCommitHistory,
         BuildItems,
         BuildCommits,
+        MicrosoftGraphCache,
```
Comment (OsmondJiang) [inline src/docfx/lib/log/TelemetryName.cs]: MicrosoftGraphAlias

File: src/docfx/lib/msgraph/MicrosoftGraphCache.cs [hunk 3]
```diff
@@ -43,57 +41,52 @@ public MicrosoftGraphCache(Config config)
...
+            {
+                return (null, msAlias);
+            }
 
-            async Task<(Error error, MicrosoftAlias msAlias)> GetAsyncCore()
+            if (_tempInvalidAliases.Contains(alias) || _microsoftGraphAccessor == null)
```
Comment (OsmondJiang) [inline src/docfx/lib/msgraph/MicrosoftGraphCache.cs]: ConcurrentBag

Comment (yufeih) [reply src/docfx/lib/msgraph/MicrosoftGraphCache.cs]: @Osmond, for each over concurrentbag creates a snapshot of all items, Contains method is using LINQ, everytime contains is called, the collection is duplocated. We have ConcurrentHashset for this purpose

Comment (OsmondJiang) [reply src/docfx/lib/msgraph/MicrosoftGraphCache.cs]: I have offline talked with @LowEntropyBody , he should use ConcurrentHashset

**Pull request closed without merging**